nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -359,31 +362,30 @@ def add_electrodes(
             if nwbfile.electrodes is None:
                 nwbfile.add_electrode_column(name=name, description=des_args["description"], index=des_args["index"])
             else:
-                samp_data = des_dict["data"][0]
-                data_type_found = [proptype for proptype in property_default_types if isinstance(samp_data, proptype)][
-                    0
-                ]
-                extended_data = [property_default_types[data_type_found]] * len(nwbfile.electrodes.id)
-                previous_data = des_args["data"].tolist()
-                if data_type_found == Real:
-                    previous_data = [float(x) for x in previous_data]
-                des_args["data"] = extended_data + previous_data
                 elec_columns_append[name] = des_args
 
     for name in elec_columns_append:
         _ = elec_columns.pop(name)
 
-    for j, channel_id in enumerate(checked_recording.get_channel_ids()):
-        if channel_id not in nwb_elec_ids:
+    channel_name_array = checked_recording.get_channel_ids()
+    # If the channel ids are integer keep the old behavior of asigning electrodes.ids equal to channel_ids
+    if np.issubdtype(channel_name_array.dtype, np.integer):
+        elec_columns["id"].update(data=channel_name_array)
+
```

Comment:
I would have imagined a coerced RecordingExtractor -> BaseRecording object would have its IDs upcast to strings. Is this not the case? If so, I do think we should probably adjust that in the `OldtoNew` functions. @alejoe91

Reply:
This is not the case and if we do this some of our data tests will fail with the current implementation that I have on this PR.

The reason being is that when the channel_ids are strings I assigned nwb_file.elctrode.ids to be `range(number_of_electrodes)` but one of our test in data (I don't remember which one) has channel_ids that are like this [1, 2, 3, 4, 125, 126] (integers). So if you cast to string the current implementation assigns nwb_file.electrode.ids as [0, 1, 2, 3, 4, 5, 6] and the comparison of channel_ids of the original recorder to the channel_ids of the nwb_recorder fails. 

One way around would be make to use a function like it `isdigit()` to with the strings for the cases where the strings channel_ids can be easily assigned as the nwb_file.electrode.ids but then I am not sure we are gaining something. What do you think?

Reply:
@CodyCBakerPhD We don't enforce any specific dtype on the SI side (and actually most of the extractors still use int). So we have to deal with both str and ints here

Reply:
Might as well force the upcasting ourselves then. (a) always include `channel_name`, (b) always have it be `str(channel_id)`